### DISCUSSION TITLE
How to debug using vite-node in VSCode?

### DISCUSSION THREAD

**Discussion by cjnoname** [vitest-dev/vitest#6096] [Q&A]
Hi,

We plan to switch from ts-node to vite-node.

Currently, we have set up the following in our vscode settings:
"debug.javascript.autoAttachFilter": "smart"

It works well with Nodemon + ts-node, automatically adding the debugger when running a script.

We've tried several times to achieve the same goal with vite-node but haven't had any luck.

Could any experts help with this?

Additionally, we want to run ESLint as part of the watch restart. Any ideas?

Thanks!

**Comment by sheremet-va** [comment]
Why do you want to switch to `vite-node` specifically? If you don't rely on custom Vite plugins, I would recommend using [`tsx`](https://www.npmjs.com/package/tsx) instead. `vite-node` is softly deprecated, and we are considering deprecating it fully after Vite 6 is released since its API is migrated there.

**Reply by scamden** [reply]
does tsx also support HMR?

**Comment by kp-eugene-kulabuhovs** [comment]
Here's an example that worked for me:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Program",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "program": "${workspaceFolder}/node_modules/vite-node/dist/cli.mjs",
            "args": ["${workspaceFolder}/src/main.ts"],
            "outFiles": [
                "${workspaceFolder}/**/*.js"
            ]
        }
    ]
}
```

**Reply by btoo** [reply]
worked like a charm

**Comment by zhaoxuyll** [comment]
node --inspect=${process.env.INSPECT_PORT} ./node_modules/vite/bin/vite.js serve

silky
